src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs

```diff
@@ -3,15 +3,13 @@
 // See the LICENSE file in the project root for more information.
```

Comment:
this type is effectively rewritten.  it's similar in general approach to AbstractDocumentBasedFixAllProvider in https://github.com/dotnet/roslyn/pull/50209, but customized for the limitations of the normal BatchFixer (namely that we can't known when we are done with a document until we compute all fixes).